src/user/service.ts

```diff
@@ -117,3 +118,70 @@ export async function validateUserCredentials(user: User, password: string, log:
   return true;
 }
 
+/**
+ * @description - Given a principal User, uses the principal id to find the UserInfo record for a user.
+ * @param user - The principal User to find the UserInfo record for.
+ * @returns The UserInfo record for the user.
+ * @throws NotFound - If the UserInfo record is not found.
+ */
+export async function findUserInfoById(user: User): Promise<UserInfo> {
+
+  const result = await db('user_info')
+      .select()
+      .where({principal_id: user.id})
+      .first();
+
+  if (!result) throw new NotFound(`UserInfo for user "${user.id}" not found.`);
+
+  return toUserInfo(user, result);
+}
+
+/**
+ * @description - Given a principal User, uses the principal id to find the Principal user and update the UserInfo record for a user.
+ * @param user - The principal User to update the UserInfo record for.
+ * @param userInfo - new UserInfo object to update the UserInfo record with.
+ * @returns The updated UserInfo record.
+ * @throws BadRequest - If the UserInfo record is not updated.
+ */
+export async function updateUserInfo(user: User, userInfo: UserInfo): Promise<UserInfo> {
+  
+  const result = await db('user_info')
+    .where({principal_id: user.id})
+    .update(userInfo)
+    .returning('*')
+    .first();
+
+  if (!result) throw new BadRequest(`UserInfo for user "${user.id}" was not updated.`);
+
+  return toUserInfo(user, result);
+}
+
+export type UserInfoModel = {
+  sub: string;
```

Comment:
`sub` should not exist. Did it accidentally make it in the database model?

Reply:
Nope... did not make it in there. 
Mental model was twisted as I believed this model had to include everything already required in our JSON response

Reply:
twas not on the migration script, but hmmmm
<img width="1268" alt="Screenshot 2025-03-05 at 11 28 50 PM" src="https://github.com/user-attachments/assets/0b908e0f-9774-4ef6-9aa0-aa2fd16acd71" />